local-server/local-server-services/src/main/java/com/esri/samples/localserver/LocalServerServicesController.java

```diff
@@ -0,0 +1,287 @@
+
+public class LocalServerServicesController {
+
+  @FXML
+  private Button btnStartServer;
+  @FXML
+  private Button btnStopServer;
+  @FXML
+  private Button btnStartService;
+  @FXML
+  private Button btnStopService;
+  @FXML
+  private Button btnURL;
+  @FXML
+  private ComboBox<String> serviceOptions;
+  @FXML
+  private TextArea centerText;
+  @FXML
+  private ListView<String> runningServices;
+
+  private boolean isServerStopping;
+
+  private LocalServer server;
+  private ListenableList<LocalService> services;
+  private HostServices hostServices;
+
+  /**
+   * Handles starting and stopping a local server.
+   * 
+   * @param event the event that fired this action
+   */
+  @FXML
+  protected void handleStartStopLocalServer(ActionEvent event) {
+
+    // find out what fired this event
+    EventTarget eventTarget = event.getTarget();
+
+    // starting server
+    if (eventTarget.equals(btnStartServer)) {
+      // start local server
+      server = LocalServer.INSTANCE;
+      isServerStopping = false;
+      server.addStatusChangedListener(status -> {
+        centerText.appendText("Server Status: " + status.getNewStatus().toString() + "\n");
+      });
+      server.startAsync();
+
+      // tracking services attached to this local server
+      services = server.getServices();
+
+      // update buttons on layout
+      btnStartServer.setDisable(true);
+      btnStopServer.setDisable(false);
+      btnStartService.setDisable(false);
+
+      // stopping server
+    } else if (eventTarget.equals(btnStopServer)) {
+      terminate();
+      btnStartServer.setDisable(false);
+      btnStopServer.setDisable(true);
+      btnStartService.setDisable(true);
+      btnStopService.setDisable(true);
+    }
+  }
+
+  /**
+   * Gets the selected item from the combo box and creates a Local Service to add to the Local Server. 
+   * <p>
+   * Only creates a Local Service that is not currently  running on the Local Server.
+   * 
+   * @param event the event that fired this action
+   * @throws URISyntaxException if URL passed to Local Service is incorrect
+   */
+  @FXML
+  protected void handleStartService(ActionEvent event) throws URISyntaxException {
+
+    if (server.getStatus() == LocalServerStatus.STARTED) {
+      // check that the service selected is not already running
+      boolean serviceExists = false;
+      String serviceSelected = serviceOptions.getSelectionModel().getSelectedItem();
+      for (String runningService : runningServices.getItems()) {
+        if (runningService.startsWith(serviceSelected)) {
+          serviceExists = true;
+        }
+      }
+
+      // start service if not currently running on the Local Server
+      if (!serviceExists) {
+        String serviceUrl = "";
+        LocalService localService = null;
+        switch (serviceSelected) {
+          case "Map Service":
+            serviceUrl = Paths.get(getClass().getResource("/PointsofInterest.mpk").toURI()).toString();

```

Comment:
This doesn't work when run from a jar (sample viewer). We'll have to add these files to samples-data when this PR goes in: https://github.com/Esri/arcgis-runtime-samples-java/pull/67/files. For now, it's fine though.